linux-pam: 1.6.1 -> 1.7.1

LordGrimmauld opened the pull request:
PAM now uses meson to build, so the package needed to essentially be rewritten.
PAM also now uses logind for login limits. To not get infinite recursion, PAM was first stripped from `systemdMinimal`, then `systemdMinimal` was used as logind lib.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LordGrimmauld committed:
systemd: disable pam in transitive dependencies if pam is disabled

LordGrimmauld committed:
kbd: allow building without pam

arianvp commented on pkgs/by-name/li/libeconf/package.nix:
```diff
@@ -0,0 +1,40 @@
+{
```
Can we build `linux-pam` with `econf` support disabled instead of introducing this package?  We don't use `econf` anywhere else in nixpkgs as the NixOS module system kind of makes it an unnecessary abstraction as the module system already implements config merging.  I'd avoid introducing it if there's no clear need.

LordGrimmauld replied:
Good call:

```
linux-pam> Linux-PAM 1.7.1
linux-pam>
linux-pam>   User defined options
linux-pam>     audit        : enabled
linux-pam>     auto_features: auto
linux-pam>     bindir       : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/bin
linux-pam>     buildtype    : plain
linux-pam>     elogind      : disabled
linux-pam>     examples     : false
linux-pam>     includedir   : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/include
linux-pam>     infodir      : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/share/info
linux-pam>     libdir       : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/lib
linux-pam>     libexecdir   : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/libexec
linux-pam>     localedir    : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/share/locale
linux-pam>     logind       : enabled
linux-pam>     mandir       : /nix/store/pn96n0wgsd7wzs8nyipvpw0mmjm2s4ls-linux-pam-1.7.1-man/share/man
linux-pam>     nis          : disabled
linux-pam>     pam_lastlog  : enabled
linux-pam>     pam_unix     : enabled
linux-pam>     prefix       : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1
linux-pam>     sbindir      : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/sbin
linux-pam>     selinux      : disabled
linux-pam>     sysconfdir   : etc
linux-pam>     wrap_mode    : nodownload
linux-pam>     xtests       : false
```

`logind` is listed as `enabled`, so this should be fine. Bonus: We can now use the github fetcher too! I'll have to rerun the test, but this should be fine.

LordGrimmauld committed:
linux-pam: re-enable doc/man outputs

Co-Authored-by: Marcin Serwin <marcin@serwin.dev>

LordGrimmauld commented on pkgs/by-name/li/linux-pam/package.nix:
Good call (wrong review comment):

```
linux-pam> Linux-PAM 1.7.1
linux-pam>
linux-pam>   User defined options
linux-pam>     audit        : enabled
linux-pam>     auto_features: auto
linux-pam>     bindir       : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/bin
linux-pam>     buildtype    : plain
linux-pam>     elogind      : disabled
linux-pam>     examples     : false
linux-pam>     includedir   : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/include
linux-pam>     infodir      : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/share/info
linux-pam>     libdir       : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/lib
linux-pam>     libexecdir   : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/libexec
linux-pam>     localedir    : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/share/locale
linux-pam>     logind       : enabled
linux-pam>     mandir       : /nix/store/pn96n0wgsd7wzs8nyipvpw0mmjm2s4ls-linux-pam-1.7.1-man/share/man
linux-pam>     nis          : disabled
linux-pam>     pam_lastlog  : enabled
linux-pam>     pam_unix     : enabled
linux-pam>     prefix       : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1
linux-pam>     sbindir      : /nix/store/fpjjk1xj0vyb3m3m2cslbvh8xxlh088s-linux-pam-1.7.1/sbin
linux-pam>     selinux      : disabled
linux-pam>     sysconfdir   : etc
linux-pam>     wrap_mode    : nodownload
linux-pam>     xtests       : false
```

`logind` is listed as `enabled`, so this should be fine. Bonus: We can now use the github fetcher too! I'll have to rerun the test, but this should be fine.

arianvp commented on pkgs/by-name/li/linux-pam/package.nix:
Should we file a ticket for this?

LordGrimmauld replied:
I opened https://github.com/NixOS/nixpkgs/issues/429075

SuperSandro2000 reviewed (commented):
just a small side node

LordGrimmauld committed:
systemd: Don't build nspawn for systemdLibs

LordGrimmauld commented on pkgs/os-specific/linux/systemd/default.nix:
```diff
@@ -255,6 +258,21 @@ stdenv.mkDerivation (finalAttrs: {
     ./0019-meson-Don-t-link-ssh-dropins.patch
 
     ./0020-install-unit_file_exists_full-follow-symlinks.patch
+
+    # add nspawn build option flag
+    # required to disable nspawn for systemdLibs to avoid dependency on getent
+    # https://github.com/systemd/systemd/pull/36876, remove for systemd 258
+    (fetchpatch {
+      # required for the actual patch to apply
+      url = "https://github.com/systemd/systemd/commit/b1fb2d971c810e0bdf9ff0ae567a1c6c230e4e5d.patch";
+      hash = "sha256-JBheazg1OFkx8vUl2l8+34BoEPVURBQJHxqntOBYB60=";
+      includes = [ "src/nspawn/meson.build" ];
+    })
+    (fetchpatch {
+      url = "https://github.com/systemd/systemd/commit/d95818f5221d9b9b19648cffa0cb2407f023b27e.patch";
+      hash = "sha256-FTpWGec5ivlkyEEDMCPaLE+BH91e7JI0kH8pS88bBDY=";
+      excludes = [ "test/fuzz/meson.build" ];
+    })
```
TODO: wait for https://github.com/NixOS/nixpkgs/pull/427968

nikstur replied:
Maybe we just disable the logind support for now in pam on musl on since that wasn't present before either and re-enable it once we have the new systemd version.

LordGrimmauld replied:
utmp has to die, and if logind is the replacement, we should enable it. pam was/is unmaintained in nixpkgs, it was long overdue someone actually took a look at it. As it stands, this PR is a full "lets do it properly" PR.
I also don't see the issue with patching in the nspawn toggle: our `systemdLibs` does not *install* nspawn anyways, so we might as well not build it either. This does not actually change anything in the package output, so i am not sure what the fuss is about.

LordGrimmauld commented:
https://github.com/NixOS/nixpkgs/issues/404169
indeed seems this was broken before. So yes, we should probably switch to `lastlog2` in hopes of also fixing musl. I suppose lastlog not being built is fine then. We still don't want the inf rec, but disabling lastlog on musl seems reasonable then.

LordGrimmauld committed:
linux-pam: 1.6.1 -> 1.7.1

PAM switched to meson build upstream, which requires some care:
- build with logind support
- reenable lastlog
- clean up autotools remnants
- disable debug as it slows down test making it fail
